### ISSUE TITLE
problem refreshing image files

### ISSUE THREAD

**Issue by robinflatland** [JuiceMind/JuiceMind#323]
Hi,

I am having a refresh problem with images displayed in Juicemind.  In our multimedia curriculum, students frequently write code that reads in an image, modifies it, and then writes the modified image out to a file.  The problem is that Juicemind is not refreshing the image in the display -- it still shows the old version of the image, even though the image file has been overwritten with a new version.   Just to make sure I am communicating the problem clearly, I made a short video showing what is happening:

https://youtu.be/PXqa7JEnRrI

With the display of the image not being automatically refreshed, the students are going to think their code didn't work.

Should you need it, this code is in my DevelopmentCSIS110 team under Lab 1

Thank you!

Robin
@juicemind-dev

**Comment by juicemind-dev** [comment]
Hi @robinflatland we're looking into this now. Thanks for the video.

**Comment by robinflatland** [comment]
Thank you for your quick attention to this.  We have a group of teachers
coming in for training in Juicemind on Monday.  They will be working
through this lab using images.  Any chance it might be working by Monday?

Robin

**Comment by juicemind-dev** [comment]
@robinflatland yes I believe we will almost certainly have this fixed by Monday. We will update you soon on this one. Hopefully some time today.

**Comment by juicemind-dev** [comment]
Hi @robinflatland this issue should be resolved now. Let me know if you have any additional problems with it.

**Comment by robinflatland** [comment]
Hi,

Thank you for your quick work on this!! It is working, and just in time for
our teaching training this morning:-)

Before we close this out, there is one thing I notice that I think is
probably easy to correct.  If the output.jpg file already exists, and the
student is looking at it in a tab, and then they press the run button, the
tab showing the image doesn't refresh automatically when the output.jpg
file is overwritten.  The student has to first switch to a different tab
and then change back to the output.jpg tab, and then it refreshes.

What would be good is if whenever they are looking at output.jpg in a tab,
it is the current version of the file. Would it be possible to refresh all
output.jpg tabs whenever output.jpg is overwritten?

Robin

**Comment by ProfGoldstein** [comment]
Dear @juicemind-dev 

I just tried the same thing, but with wave files ( wavfile.write(filename, JESsnd.sampleRate, JESsnd.samples) ) and the sound files are not updating after the initial run,  just like the problem noted above with images. 

Not sure if we can add to this issue or if you want to open a new ticket.  Let me know
Thanks
--Ira

**Comment by juicemind-dev** [comment]
@ProfGoldstein No, let's keep this issue open. We're working on a fix right now that will also increase the general speed of the platform and fix this. It will hopefully be released some time today.

**Comment by juicemind-dev** [comment]
@robinflatland Yes, this is definitely the ideal behavior. We will work on a fix for that now.

**Comment by robinflatland** [comment]
Hi,

I want to follow up on this tab refresh issue as we approach the start of the school year next week:

> If the output.jpg file already exists, and the
> student is looking at it in a tab, and then they press the run button, the
> tab showing the image doesn't refresh automatically when the output.jpg
> file is overwritten.  The student has to first switch to a different tab
> and then change back to the output.jpg tab, and then it refreshes.

When do think a fix for this might come out?  

THank you!

Robin
@ProfGoldstein 
@juicemind-dev

**Comment by marklee45** [comment]
@robinflatland 

I'm so sorry for the late reply on this!! We're coming out with a big patch to our IDE that will significantly improve compiling times, with this fix, the issue mentioned here should also be fixed. That will be coming out early this week.

**Comment by robinflatland** [comment]
Thank you! Looking forward to the improved IDE patch!

Robin
